v2/api/appconfiguration/v20240601/structure.txt

```diff
@@ -0,0 +1,282 @@
+// Code generated by azure-service-operator-codegen. DO NOT EDIT.
+github.com/Azure/azure-service-operator/v2/api/appconfiguration/v20240601
+-------------------------------------------------------------------------
+APIVersion: Enum (1 value)
+└── "2024-06-01"
+
+ConfigurationStore: Resource
+├── Owner: resources/v1apiv20191001.ResourceGroup
+├── Spec: Object (16 properties)
+│   ├── AzureName: Validated<string> (3 rules)
+│   │   ├── Rule 0: MaxLength: 50
+│   │   ├── Rule 1: MinLength: 5
+│   │   └── Rule 2: Pattern: "^[a-zA-Z0-9_-]*$"
+│   ├── CreateMode: *Enum (2 values)
+│   │   ├── "Default"
+│   │   └── "Recover"
+│   ├── DataPlaneProxy: *Object (2 properties)
+│   │   ├── AuthenticationMode: *Enum (2 values)
+│   │   │   ├── "Local"
+│   │   │   └── "Pass-through"
+│   │   └── PrivateLinkDelegation: *Enum (2 values)
+│   │       ├── "Disabled"
+│   │       └── "Enabled"
+│   ├── DefaultKeyValueRevisionRetentionPeriodInSeconds: *int
+│   ├── DisableLocalAuth: *bool
+│   ├── EnablePurgeProtection: *bool
+│   ├── Encryption: *Object (1 property)
+│   │   └── KeyVaultProperties: *Object (2 properties)
+│   │       ├── IdentityClientId: *string
+│   │       └── KeyIdentifier: *string
+│   ├── Identity: *Object (2 properties)
+│   │   ├── Type: *Enum (4 values)
+│   │   │   ├── "None"
+│   │   │   ├── "SystemAssigned"
+│   │   │   ├── "SystemAssigned, UserAssigned"
+│   │   │   └── "UserAssigned"
+│   │   └── UserAssignedIdentities: Object (1 property)[]
+│   │       └── Reference: genruntime.ResourceReference
+│   ├── Location: *string
+│   ├── OperatorSpec: *Object (3 properties)
+│   │   ├── ConfigMapExpressions: *core.DestinationExpression[]
+│   │   ├── SecretExpressions: *core.DestinationExpression[]
+│   │   └── Secrets: *Object (12 properties)
+│   │       ├── PrimaryConnectionString: *genruntime.SecretDestination
+│   │       ├── PrimaryKey: *genruntime.SecretDestination
+│   │       ├── PrimaryKeyID: *genruntime.SecretDestination
+│   │       ├── PrimaryReadOnlyConnectionString: *genruntime.SecretDestination
+│   │       ├── PrimaryReadOnlyKey: *genruntime.SecretDestination
+│   │       ├── PrimaryReadOnlyKeyID: *genruntime.SecretDestination
+│   │       ├── SecondaryConnectionString: *genruntime.SecretDestination
+│   │       ├── SecondaryKey: *genruntime.SecretDestination
+│   │       ├── SecondaryKeyID: *genruntime.SecretDestination
+│   │       ├── SecondaryReadOnlyConnectionString: *genruntime.SecretDestination
+│   │       ├── SecondaryReadOnlyKey: *genruntime.SecretDestination
+│   │       └── SecondaryReadOnlyKeyID: *genruntime.SecretDestination
+│   ├── Owner: *genruntime.KnownResourceReference
+│   ├── PublicNetworkAccess: *Enum (2 values)
+│   │   ├── "Disabled"
+│   │   └── "Enabled"
+│   ├── Sku: *Object (1 property)
+│   │   └── Name: *string
+│   ├── SoftDeleteRetentionInDays: *int
+│   ├── SystemData: *Object (6 properties)
```

Comment:
I think we can prune this whole thing as it's supposed to be readonly?

For example we do:
```
  - group: kubernetesconfiguration
    name: FluxConfiguration_Spec;FluxConfiguration_STATUS
    property: SystemData
    remove: true
    because: These properties should be marked readonly
```